scripts/anvil_tools/set_up_anvil_workspaces.py

```diff
@@ -0,0 +1,305 @@
+    response = requests.post(uri, headers=headers)
+    status_code = response.status_code
+
+    # print success or fail message based on status code
+    if status_code == 201:
+        print(f"Successfully created Authorization Domain with name: {auth_domain_name}.")
+        return status_code
+    elif status_code in ["403", "409"]:
+        print(f"WARNING: Failed to create Authorization Domain with name: {auth_domain_name}.")
+        return "Authorization Domain with name already exists. Try again with unique group name."
+    else:
+        print(f"WARNING: Failed to create Authorization Domain with name: {auth_domain_name}.")
+        print("Check output file for error details.")
+        return response.json()
+
+
+def setup_single_workspace(workspace, project="anvil-datastorage"):
+    """Create one workspace and set up with authorization domain and ACLs."""
+
+    # initialize workspace dictionary with default values assuming failure
+    workspace_dict = {"input_workspace_name": "NA",
+                      "input_auth_domain_name": "NA",
+                      "auth_domain_email": "Incomplete",
+                      "auth_domain_creation_error": "NA",
+                      "email_added_to_AD": "Incomplete",
+                      "add_email_to_AD_error": "NA",
+                      "workspace_link": "Incomplete",
+                      "workspace_creation_error": "NA",
+                      "workspace_ACLs": "Incomplete",
+                      "workspace_ACLs_error": "NA",
+                      "workspace_setup_status": "Failed"}
+
+    # start authorization domain
+    auth_domain_name = workspace['auth_domain_name']
+    workspace_dict["input_auth_domain_name"] = auth_domain_name
+    auth_domain_status = create_auth_domain(auth_domain_name)
+
+    if auth_domain_status != 201:  # AD creation fail - abort setup
+        workspace_dict["auth_domain_creation_error"] = auth_domain_status  # update error in dict
+        return workspace_dict
+
+    # AD creation success - continue setup
+    workspace_dict["auth_domain_email"] = f"{auth_domain_name}@firecloud.org"  # update dict with AD email
+    # add members to AD
+    anvil_admin_group_name = "anvil-admins@firecloud.org"
+    add_member_status = add_member_to_auth_domain(auth_domain_name, anvil_admin_group_name, "ADMIN")
+
+    if add_member_status != 204:  # adding member to AD fail - abort setup
+        workspace_dict["add_email_to_AD_error"] = auth_domain_status  # update error in dict
+        return workspace_dict
+
+    # adding member to AD success - continue setup
+    workspace_dict["email_added_to_AD"] = anvil_admin_group_name  # update dict with member added to AD
+
+    # workspace creation if AD set up succeeds
+    workspace_name = workspace["workspace_name"]
+    workspace_dict["input_workspace_name"] = workspace_name
+    create_ws_request = make_create_workspace_request(workspace_name, auth_domain_name, project)  # json for API request
```

Comment:
I'd move this entire call to inside `create_workspace` - you don't need to create the json before calling that function. it would allow you to make the inputs to `create_workspace` more intuitive (it's sort of strange for it to take in a json-formatted-request from the start - that's a hard thing to program around if you're going to reuse it) - you could have it take in workspace_name, project, and auth_domain_name